dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java

```diff
@@ -264,36 +264,64 @@ protected static Logger createLogger(final Class<?> cls,
                 try {
                     b = BundleUtils.getBundle(cls, bundleName);
                 } catch (MissingResourceException rex) {
-                    //ignore
                 }
             }
             if (b != null) {
                 b.getLocale();
             }
 
             if (loggerClass != null) {
-                try {
-                    Constructor<?> cns = loggerClass.getConstructor(String.class, String.class);
-                    if (name == null) {
+                // Default logger of Liberty is WsLogger. 
+                // CXFLogger is inherited from this WsLogger class to preserve same behavior
+                // except the one that we aimed to change. Which is to add class name to log records
+                if (loggerClass.equals(WsLogger.class)) {
+                    try {
+                        Constructor<?> cns = loggerClass.getConstructor(String.class, String.class, Class.class); //Liberty change: Class is added as parameter
+                        if (name == null) {
+                            try {
+                                return (CXFLogger) cns.newInstance(loggerName, bundleName, cls); //Liberty change: Class is added as parameter
+                            } catch (InvocationTargetException ite) {
+                                if (ite.getTargetException() instanceof MissingResourceException) {
+                                    return (CXFLogger) cns.newInstance(loggerName, null, cls); //Liberty change: Class is added as parameter
+                                }
+                                throw ite;
+                            }
+                        }
                         try {
-                            return (Logger) cns.newInstance(loggerName, bundleName);
+                            return (CXFLogger) cns.newInstance(loggerName, bundleName, cls); //Liberty change: Class is added as parameter
                         } catch (InvocationTargetException ite) {
                             if (ite.getTargetException() instanceof MissingResourceException) {
-                                return (Logger) cns.newInstance(loggerName, null);
+                                throw (MissingResourceException) ite.getTargetException();
```

Comment:
Even though this is s a separate try catch, it still needs to be consistence

```suggestion
                                return (CXFLogger) cns.newInstance(loggerName, null, cls); //Liberty change: Class is added as parameter
```